### PR TITLE
Working

### PR DESCRIPTION
**Pull request by joel**

### CODE REVIEW

File: lib/csv_row_model/export/attributes.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        #
+        # @param cell [Object] the cell's value
+        def format_cell(cell)
+          cell
+        end
+      end
+
+    end
+  end
+end
```
Comment (s12chung) [inline lib/csv_row_model/export/attributes.rb]: we can probably just move this code to `export.rb`

File: lib/csv_row_model/export/attributes.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module CsvRowModel
+  module Export
+    module Attributes
+      extend ActiveSupport::Concern
+
+      class_methods do
+        # Safe to override. Method applied to each cell by default
+        #
+        # @param cell [Object] the cell's value
+        def format_cell(cell)
```
Comment (s12chung) [inline lib/csv_row_model/export/attributes.rb]: let's change the method signature to be `format_cell(cell, column_name, column_index)` to be consistent... and just incase we want logic relating the `column_name/column_index`

File: lib/csv_row_model/import.rb [hunk 3]
```diff
@@ -36,7 +38,7 @@ def initialize(source_row, options={})
     # @return [Hash] a map of `column_name => source_row[index_of_column_name]`
     def mapped_row
       return {} unless source_row
-      @mapped_row ||= self.class.column_names.zip(source_row).to_h
+      @mapped_row ||= self.class.column_names.zip(source_row).to_h.with_indifferent_access
```
Comment (s12chung) [inline lib/csv_row_model/import.rb]: :+1:

File: lib/csv_row_model/model/dynamic_columns.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      class_methods do
+        def dynamic_columns_headers(context={})
+          dynamic_column_names.map do |column_name|
+            OpenStruct.new(context).public_send(column_name).each do |header_model|
+              public_send("#{column_name.to_s.singularize}_header", header_model)
+            end
+          end.flatten
+        end
+
+        VALID_OPTIONS_KEYS = %i[request]
```
Comment (s12chung) [inline lib/csv_row_model/model/dynamic_columns.rb]: delete :P (my bad, missed it)

File: lib/csv_row_model/model/dynamic_columns.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+        protected
+
+        def merge_dynamic_columns(column_hash)
+          @_dynamic_columns ||= {}
+          deep_clear_class_cache(:@_dynamic_columns)
+          @_dynamic_columns.merge!(column_hash)
+        end
+
+        def dynamic_column(column_name, options={})
```
Comment (s12chung) [inline lib/csv_row_model/model/dynamic_columns.rb:59]: guess we don't have any options

File: spec/csv_row_model/export/cell_format_spec.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe CsvRowModel::Export::File do
+
+  class CellFormatRowModel
+    include CsvRowModel::Model
+
+    column :first_name, header: 'First Name'
+    column :last_name,  header: 'Last Name'
+  end
```
Comment (s12chung) [inline spec/csv_row_model/export/cell_format_spec.rb]: refactor this and tests to match file it is testing.

**Comment by s12chung** [comment]
it's all working :)

3 main things for polish really....

## Ensure defining a method is called when `column` or `dynamic_column` is called

This is fixed on `import`, but I never got to it on `export` and `dynamic_column`

this is the test that replicates the problem: https://github.com/FinalCAD/csv_row_model/blob/260ef64e6fc066141de61be6797819ba5bf9dcf4/spec/csv_row_model/import/attributes_spec.rb#L99-L99

i do this by defining the attribute method separately:
https://github.com/FinalCAD/csv_row_model/blob/11b88b92ad7bc8d80e3d6e81b5895b8a38e71dc9/lib/csv_row_model/import/attributes.rb#L85-L85

and calling it in 2 places:
need to be done for `export/dynamic_column`: https://github.com/FinalCAD/csv_row_model/blob/bec15677c48bf38b9814de2b836a590dfb709f65/lib/csv_row_model/import.rb#L103-L103
replace with similar implementation to this: https://github.com/FinalCAD/csv_row_model/blob/bec15677c48bf38b9814de2b836a590dfb709f65/lib/csv_row_model/import.rb#L17-L17

## Memoize the `dynamic_column` for import

https://github.com/FinalCAD/csv_row_model/pull/36/files#diff-57dd5fc815f76051d510899e200fa986R24

please override these methods: https://github.com/FinalCAD/csv_row_model/blob/11b88b92ad7bc8d80e3d6e81b5895b8a38e71dc9/lib/csv_row_model/import/attributes.rb#L22-L46

also... the `attributes` method should include the `dynamic_column`

## Make tests into detailed test cases

The new tests are more like integration tests---rather have them very detailed test cases....

File: lib/csv_row_model/model/dynamic_columns.rb (hunk 5 above)
Comment (joel) [inline lib/csv_row_model/model/dynamic_columns.rb:59]: It's true but I leave mechanism here, if we add one should be easier

**Pull request merged**